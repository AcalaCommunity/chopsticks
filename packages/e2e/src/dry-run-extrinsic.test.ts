import { describe, expect, it } from 'vitest'

import { api, chain, dev, env, setupApi, testingPairs } from './helper'

setupApi({
  ...env.mandala,
  mockSignatureHost: true,
})

describe('dry-run-extrinsic', () => {
  it('dry run extrinsic', async () => {
    const properties = await chain.api.chainProperties
    const { alice, bob } = testingPairs('ed25519', properties.ss58Format)

    await dev.setStorage({
      System: {
        Account: [[[alice.address], { providers: 1, data: { free: 1000 * 1e12 } }]],
      },
    })
    const extrinsic = await api.tx.balances.transfer(bob.address, 1e12).signAsync(alice)
    const { outcome, storageDiff } = await chain.dryRunExtrinsic(extrinsic.toHex())
    expect(outcome.toHuman()).toMatchSnapshot()
    expect(storageDiff).toMatchSnapshot()
  })

  it('dry run extrinsic with fake signature', async () => {
    const ALICE = '5F9wnAAXTdeKxprAbEvgVvP1GeonyBQeUN16ei9NkgyVh1FC'

    await dev.setStorage({ Sudo: { Key: ALICE } })

    // sudo.sudo(system.remark(0x01))
    const call = '0xff0000000401'
    const { outcome, storageDiff } = await chain.dryRunExtrinsic({ call, address: ALICE })

    expect(outcome.toHuman()).toMatchSnapshot()
    expect(storageDiff).toMatchSnapshot()
  })
})
