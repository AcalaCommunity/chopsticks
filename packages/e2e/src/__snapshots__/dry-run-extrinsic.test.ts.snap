// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dry-run-extrinsic > dry run extrinsic 1`] = `
{
  "Ok": {
    "Ok": [],
  },
}
`;

exports[`dry-run-extrinsic > dry run extrinsic 2`] = `
[
  [
    "0x26aa394eea5630e07c48ae0c9558cef70a98fdbe9ce6c55837576c60c7af3850",
    "0x0b000000",
  ],
  [
    "0x3a7472616e73616374696f6e5f6c6576656c3a",
    undefined,
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da98a437914f3dd43c1f1f4289c5f1baa47d17c2d7823ebf260fd138f2d7e27d114c0145d968b5ff5006125f2414fadae69",
    "0x000000000000000001000000000000000010a5d4e80000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  ],
  [
    "0x1da53b775b270400e7e61ed5cbc5a1464ceba562917f0ee40245104168c945b9",
    undefined,
  ],
  [
    "0x3a65787472696e7369635f696e646578",
    "0x03000000",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9280b197190fc1a763257f65171ea366e88dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee",
    "0x01000000000000000100000000000000013f6ceb918c03000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9bf9f5fbbef06491434a79dd19744ab376d6f646c6163612f63706f740000000000000000000000000000000000000000",
    "0x0000000000000000010000000000000070f800f7310000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  ],
  [
    "0x1da53b775b270400e7e61ed5cbc5a146c726478796bad0b9cabd7481dbe64983",
    undefined,
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef780d41e5e16056765bc8461851072c9d7",
    "0x2c000000000000004216c63db90f0200000001000000000002c0cd170002010000020000000a0888dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0eeff30b5e40300000000000000000000000000020000000003d17c2d7823ebf260fd138f2d7e27d114c0145d968b5ff5006125f2414fadae690000020000000a00d17c2d7823ebf260fd138f2d7e27d114c0145d968b5ff5006125f2414fadae690010a5d4e800000000000000000000000000020000000a0288dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0eed17c2d7823ebf260fd138f2d7e27d114c0145d968b5ff5006125f2414fadae690010a5d4e800000000000000000000000000020000000a076d6f646c6163612f747273790000000000000000000000000000000000000000ccc05d1d0300000000000000000000000000020000001406ccc05d1d0300000000000000000000000000020000000a076d6f646c6163612f63706f740000000000000000000000000000000000000000337057c70000000000000000000000000000020000000e0388dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0eeff30b5e4030000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000200000000002222b13e2538000000",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9ae10bf9cd0e372c142b2ca90219666906d6f646c6163612f747273790000000000000000000000000000000000000000",
    "0x00000000000000000100000000000000a9299f02820000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7df1daeb8986837f21cc5d17596bb78d19eb2dcce60f37a2702000000",
    "0x49024102840088dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee002dade78d3d31d4a250c65dc7421674c006f7801f73870021b962e208df8d27531faaaf4d1f621119c2ca217d1a04c2649df1a06aee5907dea0d938c27901f50c640000000a0000d17c2d7823ebf260fd138f2d7e27d114c0145d968b5ff5006125f2414fadae69070010a5d4e8",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef734abf5cb34d6244378cddbf18e849d96",
    "0x2222b13e2538000007b9f7338b0156550e00",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7ff553b5a9862a516939d82b3d3d8661a",
    "0x0003000000",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7a86da5a932684f199539836fcb8c886f",
    "0xab060000",
  ],
]
`;

exports[`dry-run-extrinsic > dry run extrinsic with fake signature 1`] = `
{
  "Ok": {
    "Ok": [],
  },
}
`;

exports[`dry-run-extrinsic > dry run extrinsic with fake signature 2`] = `
[
  [
    "0x26aa394eea5630e07c48ae0c9558cef70a98fdbe9ce6c55837576c60c7af3850",
    "0x07000000",
  ],
  [
    "0x3a7472616e73616374696f6e5f6c6576656c3a",
    undefined,
  ],
  [
    "0x1da53b775b270400e7e61ed5cbc5a1464ceba562917f0ee40245104168c945b9",
    undefined,
  ],
  [
    "0x3a65787472696e7369635f696e646578",
    "0x03000000",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7ff553b5a9862a516939d82b3d3d8661a",
    "0x0003000000",
  ],
  [
    "0x1da53b775b270400e7e61ed5cbc5a146c726478796bad0b9cabd7481dbe64983",
    undefined,
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef780d41e5e16056765bc8461851072c9d7",
    "0x1c000000000000004216c63db90f0200000001000000000002c0cd170002010000020000000a0888b3cb383e25bafa7195ab63bc677e1bd6cb27b97cd5d9e91e59f58bf1d3ea68ecfa1ccc020000000000000000000000000002000000ff00000000020000000a0788b3cb383e25bafa7195ab63bc677e1bd6cb27b97cd5d9e91e59f58bf1d3ea68ecfa1ccc0200000000000000000000000000020000000e0388b3cb383e25bafa7195ab63bc677e1bd6cb27b97cd5d9e91e59f58bf1d3ea6800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000200000000008a654d1800000100",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da91241e3f2e98b1a98771830bbf2ee5ee188b3cb383e25bafa7195ab63bc677e1bd6cb27b97cd5d9e91e59f58bf1d3ea68",
    "0x0500000000000000010000000000000036370e3b699e1f000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7df1daeb8986837f21cc5d17596bb78d19eb2dcce60f37a2702000000",
    "0xb901b101840088b3cb383e25bafa7195ab63bc677e1bd6cb27b97cd5d9e91e59f58bf1d3ea6801deadbeefcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcd001000ff0000000401",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef734abf5cb34d6244378cddbf18e849d96",
    "0x8a654d1800000007b9f7338b0156550e00",
  ],
  [
    "0xc2261276cc9d1f8598ea4b6a74b15c2f57c875e4cff74148e4628f264b974c80",
    "0x00806b1b0048eb1f0000000000000000",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7a86da5a932684f199539836fcb8c886f",
    "0x87060000",
  ],
]
`;
